storage/pebble/register_pruner_run.go

```diff
@@ -0,0 +1,120 @@
+package pebble
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// DBPruner is responsible for pruning outdated register entries in a Pebble database.
+// It batches deletions to improve performance and reduce the load on the system.
+type DBPruner struct {
+	logger zerolog.Logger
+	// pruneThrottleDelay controls a small pause between batches of registers inspected and pruned
+	pruneThrottleDelay time.Duration
+	pruneHeight        uint64
+
+	db                   *pebble.DB
+	lastRegisterID       flow.RegisterID
+	keepFirstRelevantKey bool
+
+	totalKeysPruned int
+}
+
+func RegisterPrunerRun(db *pebble.DB, logger zerolog.Logger, pruneThrottleDelay time.Duration, pruneHeight uint64) *DBPruner {
+	return &DBPruner{
+		logger:             logger,
+		pruneThrottleDelay: pruneThrottleDelay,
+		pruneHeight:        pruneHeight,
+		db:                 db,
+		totalKeysPruned:    0,
+	}
+}
+
+// BatchDelete deletes the provided keys from the database in a single batch operation.
+// It resets the batch for reuse, deletes each key from the batch, and commits the batch to ensure
+// that the deletions are applied atomically.
+//
+// Parameters:
+//   - ctx: The context for managing the pruning throttle delay operation.
+//   - lookupKeys: A slice of keys to be deleted from the database.
+//
+// No errors are expected during normal operations.
+func (p *DBPruner) BatchDelete(ctx context.Context, lookupKeys [][]byte) error {
+	dbBatch := p.db.NewBatch()
+	defer func() {
+		if cerr := dbBatch.Close(); cerr != nil {
+			p.logger.Err(cerr).Msg("error while closing the db batch")
+		}
+	}()
+
+	for _, key := range lookupKeys {
+		if err := dbBatch.Delete(key, nil); err != nil {
```

Comment:
Pebble batch support `DeleteRange`. 

I wonder if it would be more efficient with it. 

For instance, if we have the following registers and different height:
```
registerA-height100
registerA-height99
registerA-height98
registerA-height90
```

And we'd like to prune by height 100. 
Instead of calling:
```
dbBatch.Delete(registerA-height99, pebble.Sync)
dbBatch.Delete(registerA-height98, pebble.Sync)
dbBatch.Delete(registerA-height90, pebble.Sync)
```

we could call:
```
dbBatch.DeleteRange(registerA-height99, registerA-height0, pebble.Sync)
```

I wonder if that would make a difference?